optaplanner-core/src/main/java/org/optaplanner/core/impl/constructionheuristic/decider/MultiThreadedConstructionHeuristicDecider.java

```diff
@@ -111,12 +111,14 @@ public void phaseStarted(ConstructionHeuristicPhaseScope<Solution_> phaseScope)
     @Override
     public void phaseEnded(ConstructionHeuristicPhaseScope<Solution_> phaseScope) {
         super.phaseEnded(phaseScope);
+        // Clear non-started operations to improve shutdown performance. We're shutting down anyway, so it's fine.
+        operationQueue.clear();
         // Tell the move thread runners to stop
         DestroyOperation<Solution_> destroyOperation = new DestroyOperation<>();
         for (int i = 0; i < moveThreadCount; i++) {
             operationQueue.add(destroyOperation);
         }
-        // TODO This should probably be in a finally that spawns at least the entire phase, maybe even the entire solve
+        // TODO This should probably be in a finally that spans at least the entire phase, maybe even the entire solve
         ThreadUtils.shutdownAwaitOrKill(executor, logIndentation, "Multithreaded Local Search");
```

Comment:
FYI I went looking around the code and this may not be achievable.
Most likely the entire lifecycle class hierarchy would have to be made `AutoClosable` and the `close()` call propagated downward.